algoliasearch/client.go

```diff
@@ -48,195 +46,171 @@ func (c *Client) SetTimeout(connectTimeout int, readTimeout int) {
 
 // ListIndexes returns the list of all indexes belonging to this Algolia
 // application.
-func (c *Client) ListIndexes() (interface{}, error) {
-	return c.transport.request("GET", "/1/indexes", nil, read)
+func (c *Client) ListIndexes() (ListIndexesRes, error) {
```

Comment:
You can definitely use named return values here like `(res ListIndexesRes, err error)`. That's what I do on my PR #34. It avoids to declare them in the function and to use them all along the function. I won't put this comment on every single function of the `Client` but you get the idea. Maybe just add a commit on top of this one if you think it's worth it.